proxy/proxy.go

```diff
@@ -31,5 +32,31 @@ func (p *Proxy) WithTransport(transport *http.Transport) *Proxy {
 func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	log.Println(r.URL)
 	r.Host = p.target.Host
+	r.Header.Set("X-Forwarded-For", r.RemoteAddr) // Note: not very reliable. Maybe we should set and check X-Real-IP. Also, X-Forwarded-For might have already been set.
+
+	for _, cert := range r.TLS.PeerCertificates {
+		for _, ext := range cert.Extensions {
+			atlsVariant, err := ExtractExtensionATLSVariant(&ext)
+			if err != nil {
+				// not an ATLS variant, simply skip
+				continue
+			}
+
+			measurements, err := ExtractMeasurementsFromExtension(&ext, atlsVariant)
+			if err != nil {
+				// TODO: log this error!
+				continue // TODO: should we just terminate with 400?
+			}
+
+			marshaledPcrs, err := json.Marshal(measurements)
```

Comment:
should this forward the single-hash instead of the full measurement? i can see arguments for both ways 🤔

Reply:
Let's forward the whole measurement for now since you can always hash it as the proxied-to service

Reply:
yeah i guess you're right, better now have information loss here. and a measurement is like 800B it's well below the 4KiB nginx header size limit